cmd/crdCreator/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/client-go/dynamic"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+const baseDir string = "shim/crds"
+
+//+kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;watch;list;create
+
+func main() {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	log := ctrl.Log.WithName("main")
+
+	ctx := context.Background()
+
+	log.Info("Setting up dynamic client")
+	dynClient, err := dynamic.NewForConfig(config.GetConfigOrDie())
```

Comment:
- majorly dynamic client is created using inference of GVK from an unknown bytes
- here the use-case is different, we already know the GVK/GVR of the resource why not use a typed client, convert yaml to json and send the request, something like
``` go
// ref: https://github.com/kubernetes/apimachinery/blob/release-1.27/pkg/util/yaml/decoder.go#L108-L135
import  (
"os"
"k8s.io/apimachinery/pkg/util/yaml"
"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
)
// ..
crd := &v1.CustomResourceDefinition{}
f, _ := os.Open("YAML_FILE")
d := yaml.NewYAMLToJSONDecoder(f)
_ = d.Decode(crd)
// ..
// now you've typed struct, can create typed client-set and perform the op
```

Reply:
Updated